packages/react-aria-components/src/GridList.tsx

```diff
@@ -279,10 +287,12 @@ export interface GridListItemProps<T = object> extends RenderProps<GridListItemR
   onAction?: () => void
 }
 
+class GridListNode<T> extends ItemNode<T> {}
+
 /**
  * A GridListItem represents an individual item in a GridList.
  */
-export const GridListItem = /*#__PURE__*/ createLeafComponent('item', function GridListItem<T extends object>(props: GridListItemProps<T>, forwardedRef: ForwardedRef<HTMLDivElement>, item: Node<T>) {
+export const GridListItem = /*#__PURE__*/ createLeafComponent(GridListNode, function GridListItem<T extends object>(props: GridListItemProps<T>, forwardedRef: ForwardedRef<HTMLDivElement>, item: Node<T>) {
```

Comment:
Why bother subclassing?
```suggestion
export const GridListItem = /*#__PURE__*/ createLeafComponent(ItemNode, function GridListItem<T extends object>(props: GridListItemProps<T>, forwardedRef: ForwardedRef<HTMLDivElement>, item: Node<T>) {
```

Reply:
mainly to guard against the case where GridListNode diverges from ListBoxNode, similar to what we've experienced with Item vs Grid/ListBox/etc Item. Maybe not as necessary since these nodes aren't being exported, but future work is coming where we want to allow the user to further specify how exactly these node's `filter` function works and thus become public API (e.g. for a Tree, you may want to filter based off of what is visually available, or maybe to filter a flattened version of the Tree ignoring hierarchy, etc)